File: iOS/XtreamPlayer/Domain/UseCases/LoginUseCase.swift
// iOS/XtreamPlayer/Domain/UseCases/LoginUseCase.swift
import Foundation

protocol LoginUseCase {
    func execute(host: String, username: String, password: String) async throws -> User
}

final class LoginUseCaseImpl: LoginUseCase {
    private let repository: AuthRepository

    init(repository: AuthRepository) {
        self.repository = repository
    }

    func execute(host: String, username: String, password: String) async throws -> User {
        try await repository.login(host: host, username: username, password: password)
    }
}

File: Android/app/src/main/kotlin/com/erdin/player/domain/usecases/LoginUseCase.kt
// Android/app/src/main/kotlin/com/xtream/player/domain/usecases/LoginUseCase.kt
package com.erdin.player.domain.usecases

import com.erdin.player.domain.entities.User
import com.erdin.player.domain.repositories.AuthRepository
import javax.inject.Inject

/** Authenticates against the Xtream server and saves the profile on success. */
class LoginUseCase @Inject constructor(
    private val authRepository: AuthRepository
) {
    suspend operator fun invoke(host: String, username: String, password: String, profileName: String): Result<User> {
        if (host.isBlank() || username.isBlank() || password.isBlank()) {
            return Result.failure(IllegalArgumentException("Host, username and password are required"))
        }
        val normalizedHost = host.trim().trimEnd('/')
        val resolvedProfileName = profileName.trim().ifBlank { username.trim() }
        return authRepository.login(normalizedHost, username.trim(), password, resolvedProfileName)
    }
}
